app/controllers/hq/partners_controller.rb

```diff
@@ -5,4 +5,11 @@ class Hq::PartnersController < ApplicationController
   def index
     @partners= Partner.all
   end
+
+  def show
+    if not (@partner= Partner.find_by_id(params[:id]))
```

Comment:
still tempted to use find and let rails deal with the rare case of an incorrect id being given